test/AaveAtokenFork.t.sol

```diff
@@ -0,0 +1,174 @@
+    //     "offset": 0,
+    //     "slot": "52",
+    //     "type": "t_mapping(t_address,t_struct(ReserveData)12580_storage)"
+    //
+    // We can see here that _reserves is a mapping pointing to a struct, namely:
+    // DataTypes.ReserveData. The struct is a big one, occupying many slots. So
+    // we need to find out which slot we want. In this case, the property we
+    // care about is called "aTokenAddress" and it is stored fairly deep within
+    // the data structure. These are the properties leading up to it (see
+    // aave-v3-core/contracts/protocol/libraries/types/DataTypes.sol):
+    //
+    //   slot 1: ReserveConfigurationMap configuration; <-- just a uint256
+    //   slot 2: uint128 liquidityIndex;
+    //   slot 2: uint128 currentLiquidityRate;
+    //   slot 3: uint128 variableBorrowIndex;
+    //   slot 3: uint128 currentVariableBorrowRate;
+    //   slot 4: uint128 currentStableBorrowRate;
+    //   slot 4: uint40 lastUpdateTimestamp;
+    //   slot 4: uint16 id;
+    //   slot 5: address aTokenAddress;
+    //
+    // So we need to take the 4th slot after the one we compute for the mapping:
+    bytes32 _aTokenAddressStorageSlot = bytes32(uint256(keccak256(
+      bytes.concat(
+        bytes32(uint256(uint160(address(token)))), // map key == the token addr
+        bytes32(uint256(52)) // _reserves slot, as determined by forge
+      )
+    )) + 4); // 4 slots *after* the slot computed for the struct, i.e. the 5th slot
+
+    aToken = AToken(
+      address(uint160(uint256(
+        vm.load(address(pool), _aTokenAddressStorageSlot)
+      )))
+    );
+  }
+
+  function testFork_SetupWorked() public {
+    assertEq(ERC20(address(aToken)).symbol(), "aOptGOV");
+    assertEq(ERC20(address(aToken)).name(), "Aave V3 Optimism GOV");
+
+    // Confirm that the atoken._underlyingAsset == token
+    // forge inspect lib/aave-v3-core/contracts/protocol/tokenization/AToken.sol:AToken storage
+    //
+    //   "label": "_underlyingAsset",
+    //   "offset": 0,
+    //   "slot": "61",
+    //   "type": "t_address"
+    assertEq(
+      address(uint160(uint256(
+        vm.load(address(aToken), bytes32(uint256(61)))
+      ))),
+      address(token)
+    );
+
+    // Confirm that we can supply GOV to the aToken.
+    assertEq(aToken.balanceOf(address(this)), 0);
+
+    // Mint GOV and deposit into aave.
+    token.THIS_IS_JUST_A_TEST_HOOK_mint(address(this), 42 ether);
+    token.approve(address(pool), type(uint256).max);
+    pool.supply(
+      address(token),
+      2 ether,
+      address(this),
+      0 // referral code
+    );
+    assertEq(token.balanceOf(address(this)), 40 ether);
+    assertEq(aToken.balanceOf(address(this)), 2 ether);
+
+    // We can withdraw our GOV when we want to.
+    pool.withdraw(
+      address(token),
+      2 ether,
+      address(this)
+    );
+    assertEq(token.balanceOf(address(this)), 42 ether);
+    assertEq(aToken.balanceOf(address(this)), 0 ether);
+  }
+}
```

Comment:
Nice job demonstrating we can supply GOV 👍

At some point, we will end up writing tests that use supplied GOV as collateral, as well as borrow GOV and liquidate GOV. This might require more setup (oracles?). I'll leave it up to you if we want to add tests in this PR and exercise that functionality, or if we should tackle that later.

Reply:
Good call, I'll just do it in this PR

Reply:
Done in https://github.com/ScopeLift/flexible-voting/pull/10/commits/4612aea6e399beec6bce0594979c7921e5622cda